scripts/west_commands/runners/openocd.py

```diff
@@ -405,6 +411,20 @@ def do_attach_debug_rtt(self, command, **kwargs):
         elif command == 'rtt':
             self.print_rttserver_message()
             server_proc = self.popen_ignore_int(server_cmd)
+
+            if os_name != 'nt':
+                # Save the terminal settings
+                fd = sys.stdin.fileno()
+                new_term = termios.tcgetattr(fd)
+                old_term = termios.tcgetattr(fd)
+
+                # New terminal setting unbuffered
+                new_term[3] = new_term[3] & ~termios.ICANON & ~termios.ECHO
+                termios.tcsetattr(fd, termios.TCSAFLUSH, new_term)
+
+                # Support normal-terminal reset at exit
+                atexit.register(lambda: {termios.tcsetattr(fd, termios.TCSAFLUSH, old_term)})
```

Comment:
I think it'd be a little cleaner to move this into the `finally` block unless we really need `atexit`. Could you clarify if this is really needed?